src/integrations/ga/GaToSplit.js

```diff
@@ -130,34 +111,56 @@ export function validateIdentities(identities) {
 }
 
 /**
- * Validates if a given object is a EventData instance, and logs corresponding warnings.
+ * Checks if EventData fields (except EventTypeId) are valid, and logs corresponding warnings.
+ * EventTypeId is validated separately.
  *
  * @param {EventData} data event data instance to validate. Precondition: data != undefined
  * @returns {boolean} Whether the data instance is a valid EventData or not.
  */
-export function validateEventData(data) {
-  if (!validateEvent(data.eventTypeId, 'splitio-ga-to-split:mapper'))
+export function validateEventData(eventData) {
+  if (validateEventValue(eventData.value, 'splitio-ga-to-split:mapper') === false)
     return false;
 
-  if (validateEventValue(data.value, 'splitio-ga-to-split:mapper') === false)
-    return false;
-
-  const { properties } = validateEventProperties(data.properties, 'splitio-ga-to-split:mapper');
+  const { properties } = validateEventProperties(eventData.properties, 'splitio-ga-to-split:mapper');
   if (properties === false)
     return false;
 
-  if (data.timestamp && !isNumber(data.timestamp))
+  if (eventData.timestamp && !isFinite(eventData.timestamp))
     return false;
 
-  if (data.key && validateKey(data.key, 'splitio-ga-to-split:mapper') === false)
+  if (eventData.key && validateKey(eventData.key, 'splitio-ga-to-split:mapper') === false)
     return false;
 
-  if (data.trafficTypeName && validateTrafficType(data.trafficTypeName, 'splitio-ga-to-split:mapper') === false)
+  if (eventData.trafficTypeName && validateTrafficType(eventData.trafficTypeName, 'splitio-ga-to-split:mapper') === false)
     return false;
 
   return true;
 }
 
+const DEFAULT_EVENT_TYPE = 'event';
+const INVALID_PREFIX_REGEX = /^[^a-zA-Z0-9]+/;
+const INVALID_SUBSTRING_REGEX = /[^-_.:a-zA-Z0-9]+/g;
+/**
+ * Fixes the passed string value to comply with EventTypeId format, by removing invalid characters and truncating if necessary.
+ *
+ * @param {string} eventTypeId string value to fix.
+ * @returns {string} Fixed version of `eventTypeId`.
+ */
+export function fixEventTypeId(eventTypeId) {
+  // set a default eventTypeId if it is not a string or is an empty one.
+  if (!isString(eventTypeId) || eventTypeId.length === 0) {
+    return DEFAULT_EVENT_TYPE;
+  }
+
+  // replace invalid substrings and truncate
+  const fixed = eventTypeId
+    .replace(INVALID_PREFIX_REGEX, '')
+    .replace(INVALID_SUBSTRING_REGEX, '_')
+    .slice(0, 80);
```

Comment:
why 80?

Reply:
part of EVENT_TYPE_REGEX: no more than 80 chars.